### ISSUE TITLE
bb10 dev

### ISSUE THREAD

**Issue by mathewjpotter** [AlexeyGurevsky/bbtube#2]
hey alex, wondering if we could chat about bb10 dev im a developer, but I dont have access to the bb10 dev tools. need a basic app with api calls

**Comment by AlexeyGurevsky** [comment]
@mathewjpotter Hello.
I would start from this page (while it's still available lol): https://developer.blackberry.com/native/
It contains all the tools, samples, tutorials, specs, etc. needed for BB10 development.
But feel free to ask specific questions if you're stuck in something

**Comment by mathewjpotter** [comment]
ok ill give it a shot, is there an issue with getting a dev license?                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone on the TELUS network.                                                                                                                                                                                                                From: AlexeyGurevskySent: Saturday, October 16, 2021 11:27 AMTo: AlexeyGurevsky/bbtubeReply To: AlexeyGurevsky/bbtubeCc: Mathew; MentionSubject: Re: [AlexeyGurevsky/bbtube] bb10 dev (#2)
@mathewjpotter Hello.
I would start from this page (while it's still available lol): https://developer.blackberry.com/native/
It contains all the tools, samples, tutorials, specs, etc. needed for BB10 development.
But feel free to ask specific questions if you're stuck in something

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/AlexeyGurevsky/bbtube/issues/2#issuecomment-944976503",
"url": "https://github.com/AlexeyGurevsky/bbtube/issues/2#issuecomment-944976503",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by AlexeyGurevsky** [comment]
> ok ill give it a shot, is there an issue with getting a dev license? 

Nope, it still works, I've just checked that.
The thing is that you don't need any license or stuff like that to start the development. Just download the IDE, NDK, and, perhaps, the emulator (if you don't have a BB 10 device) and you're ready.
The only part when you need Blackberry's participation is signing your .bar-files, i.e. the apps which you are going to use/distribute. Without that signature you wouldn't be able to upload the app into App World (obviously not relevant anymore) and you're not able to sideload the app on your BB10 device.
This process (meaning signing) requires a connection to BB infrastructure, for now it works but no one knows what's going to happen in January 2022 after BB10 EOL.

**Comment by mathewjpotter** [comment]
I was reading that with the tokens, you can set the time on your phone after they expire to install the apps, any experience with that?                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone on the TELUS network.                                                                                                                                                                                                                From: AlexeyGurevskySent: Saturday, October 16, 2021 12:43 PMTo: AlexeyGurevsky/bbtubeReply To: AlexeyGurevsky/bbtubeCc: Mathew; MentionSubject: Re: [AlexeyGurevsky/bbtube] bb10 dev (#2)

ok ill give it a shot, is there an issue with getting a dev license?
Nope, it still works, I've just checked that.
The thing is that you don't need any license or stuff like that to start the development. Just download the IDE, NDK, and, perhaps, the emulator (if you don't have a BB 10 device) and you're ready.
The only part when you need Blackberry's participation is signing your .bar-files, i.e. the apps which you are going to use/distribute. Without that signature you wouldn't be able to upload the app into App World (obviously not relevant anymore) and you're not able to sideload the app on your BB10 device.
This process (meaning signing) requires a connection to BB infrastructure, for now it works but no one knows what's going to happen in January 2022 after BB10 EOL.


—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/AlexeyGurevsky/bbtube/issues/2#issuecomment-944994797",
"url": "https://github.com/AlexeyGurevsky/bbtube/issues/2#issuecomment-944994797",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by AlexeyGurevsky** [comment]
@mathewjpotter 
Yes, debug tokens are valid for 1 month, one needs to renew them to continue development. But those tokens are needed only to deploy the apps on your dev phone and to (surprisingly) debug :) 
In theory, if for some reason a debug token is not available, you can export an app to a .bar-file and sideload it (via Sachesi for instance) on each build. Not that fast and no capability to debug but this is at least something.
For now, this infrastructure does work but it can stop working anytime soon. Presumably, it can happen in January 2022

**Comment by mathewjpotter** [comment]
cool thanks for the info, im gonna put everything down in. MD file I am planning on doing, I have a php driven api on a server to connect to my self deployed version of rocket chat. ive already prototyped calls in an android app. but would like to do it in native. defn willing to pay for a little consult / skill when I dig in, thanks for your help thus far                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone on the TELUS network.                                                                                                                                                                                                                From: AlexeyGurevskySent: Sunday, October 17, 2021 10:44 AMTo: AlexeyGurevsky/bbtubeReply To: AlexeyGurevsky/bbtubeCc: Mathew; MentionSubject: Re: [AlexeyGurevsky/bbtube] bb10 dev (#2)
@mathewjpotter
Yes, debug tokens are valid for 1 month, one needs to renew them to continue development. But those tokens are needed only to deploy the apps on your dev phone and to (surprisingly) debug :)
In theory, if for some reason a debug token is not available, you can export an app to a .bar-file and sideload it (via Sachesi for instance) on each build. Not that fast and no capability to debug but this is at least something.
For now, this infrastructure does work but it can stop working anytime soon. Presumably, it can happen in January 2022

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/AlexeyGurevsky/bbtube/issues/2#issuecomment-945165844",
"url": "https://github.com/AlexeyGurevsky/bbtube/issues/2#issuecomment-945165844",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by mathewjpotter** [comment]
Good Morning Alex,

I've got everything setup and took a look at some of the app templates, 
I should be able to do most of this.

On a couple of the templates I keep getting

make[2]: *** [o.le-v7-g/databasecrud] Error 1
make[2]: Leaving directory 
'C:/Users/mp/momentics-workspace/Cascades-Samples-master/Cascades-Samples-master/databasecrud/arm'
make[1]: *** [debug] Error 2
Makefile:50: recipe for target 'debug' failed
make[1]: Leaving directory 
'C:/Users/mp/momentics-workspace/Cascades-Samples-master/Cascades-Samples-master/databasecrud/arm'
mk/cs-base.mk:31: recipe for target 'Device-Debug' failed
make: *** [Device-Debug] Error 2

basically something to do with the arm directory, i noticed it has make 
files whats the process in momentics to deal with that/fix this issue

Thanks!

also check your paypal ;)

**Comment by AlexeyGurevsky** [comment]
Hi @mathewjpotter 
Normally build error logs in Momentics are not helpful at all. And, again, normally the 'arm' folder doesn't cause any errors by itself. It's a folder where Momentics puts all executables. If you clean the project then the folder will be deleted.
I assume that some sample projects can be built without any errors on your local machine, so basically the issue can be caused by a specific project.
Could you share a link to this 'databasecrud' project that I could check it on my machine? 

As for PayPal, technically, it is not my account (long story :) ) 
I believe that you've got it from the BBTube forum thread, right?
But I'll try to check that with the account's owner.
You didn't have to do that btw, I haven't done anything crucial for you yet ;-)

**Comment by mathewjpotter** [comment]
Thanks!

well let me know your paypal for future use!

i got the database crud from this git, then just imported it

https://github.com/blackberry/Cascades-Samples/

and tried to run

**Comment by AlexeyGurevsky** [comment]
@mathewjpotter 
It builds just fine on my machine.
Here is the log:

20:40:00 **** Incremental Build of configuration Device-Debug for project databasecrud ****
make -j6 Device-Debug 
make -C .//translations -f Makefile update
make[1]: Entering directory 'C:/sources/BB/databasecrud/translations'
C:/bbndk/host_10_3_1_12/win32/x86/usr/bin/lupdate databasecrud.pro
Updating 'databasecrud.ts'...
    Found 33 source text(s) (0 new and 33 already existing)
make[1]: Leaving directory 'C:/sources/BB/databasecrud/translations'
make -C .//translations -f Makefile release
make[1]: Entering directory 'C:/sources/BB/databasecrud/translations'
C:/bbndk/host_10_3_1_12/win32/x86/usr/bin/lrelease databasecrud.pro
Updating 'C:/sources/BB/databasecrud/translations/databasecrud.qm'...
    Generated 0 translation(s) (0 finished and 0 unfinished)
    Ignored 33 untranslated source text(s)
make[1]: Leaving directory 'C:/sources/BB/databasecrud/translations'
make -C ./arm -f Makefile debug
make[1]: Entering directory 'C:/sources/BB/databasecrud/arm'
c:/bbndk/host_10_3_1_12/win32/x86/usr/bin/qmake.exe -spec ../../../../bbndk/target_10_3_1_995/qnx6/usr/share/qt4/mkspecs/blackberry-armv7le-qcc CONFIG+=debug_and_release CONFIG+=device -o Makefile ../databasecrud.pro
make -f Makefile.Debug
make[2]: Entering directory 'C:/sources/BB/databasecrud/arm'
make[2]: Nothing to be done for 'first'.
make[2]: Leaving directory 'C:/sources/BB/databasecrud/arm'
make[1]: Leaving directory 'C:/sources/BB/databasecrud/arm'

20:40:01 Build Finished (took 1s.315ms)


I don't know why it doesn't work on yours, perhaps you imported only a part of the project if that's possible.
Also go to Momentics->Project menu -> Properties -> BlackBerry API Level->Check that 'Blackberry 10 Native SDK 10.3.1' is selected

**Comment by mathewjpotter** [comment]
Is this the same when you run it on the device?

Are u on discord?

Matt




The Swift Team - SwiftGrid.net
Swift Media Group Ltd. - a partner to grow with
#200, 200 Carnegie Road.| St. Albert, AB | Canada | T8N 1H4



Secure and Confidential Communication

This e-mail message and any attachments to this e-mail are the privileged and confidential information of Swift Media Group Ltd. and are intended solely for the individual or entity named as the recipient.

If you are not the intended recipient, or the employee or agent responsible for delivering the message to the intended recipient, you are hereby notified that any dissemination, distribution or reproduction of this communication, or any attachment to this email is strictly prohibited. If you have received this communication in error, please notify the sender, and delete this message and any attachments immediately.

**Comment by mathewjpotter** [comment]
I figured it!



The Swift Team - SwiftGrid.net
Swift Media Group Ltd. - a partner to grow with
#200, 200 Carnegie Road.| St. Albert, AB | Canada | T8N 1H4



Secure and Confidential Communication

This e-mail message and any attachments to this e-mail are the privileged and confidential information of Swift Media Group Ltd. and are intended solely for the individual or entity named as the recipient.

If you are not the intended recipient, or the employee or agent responsible for delivering the message to the intended recipient, you are hereby notified that any dissemination, distribution or reproduction of this communication, or any attachment to this email is strictly prohibited. If you have received this communication in error, please notify the sender, and delete this message and any attachments immediately.

**Comment by mathewjpotter** [comment]
Hey Alex, pop by the the berryfarm os telegram,

A good group of bb10 devs

https://t.me/berryfarm_org



The Swift Team - SwiftGrid.net
Swift Media Group Ltd. - a partner to grow with
#200, 200 Carnegie Road.| St. Albert, AB | Canada | T8N 1H4



Secure and Confidential Communication

This e-mail message and any attachments to this e-mail are the privileged and confidential information of Swift Media Group Ltd. and are intended solely for the individual or entity named as the recipient.

If you are not the intended recipient, or the employee or agent responsible for delivering the message to the intended recipient, you are hereby notified that any dissemination, distribution or reproduction of this communication, or any attachment to this email is strictly prohibited. If you have received this communication in error, please notify the sender, and delete this message and any attachments immediately.
